buildkite/resource_test_suite.go

```diff
@@ -0,0 +1,252 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type testSuiteModel struct {
+	ApiToken      types.String `tfsdk:"api_token"`
+	DefaultBranch types.String `tfsdk:"default_branch"`
+	ID            types.String `tfsdk:"id"`
+	UUID          types.String `tfsdk:"uuid"`
+	TeamOwnerId   types.String `tfsdk:"team_owner_id"`
+	Name          types.String `tfsdk:"name"`
+	Slug          types.String `tfsdk:"slug"`
+}
+
+type testSuiteResponse struct {
+	ApiToken      string `json:"api_token"`
+	DefaultBranch string `json:"default_branch"`
+	UUID          string `json:"id"`
+	GraphqlID     string `json:"graphql_id"`
+	Name          string `json:"name"`
+	Slug          string `json:"slug"`
+	TeamUuids     string `json:"team_ids"`
+}
+
+type testSuiteResource struct {
+	client *Client
+}
+
+func newTestSuiteResource() resource.Resource {
+	return &testSuiteResource{}
+}
+
+func (ts *testSuiteResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	ts.client = req.ProviderData.(*Client)
+}
+
+func (ts *testSuiteResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state testSuiteModel
+	var response testSuiteResponse
+	payload := map[string]interface{}{}
+	var teamOwnerUuid string
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// The REST API requires team UUIDs but everything else in the provider uses GraphQL IDs. So we map from UUID to ID
+	// here
+	apiResponse, err := getNode(ts.client.genqlient, plan.TeamOwnerId.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to find team", err.Error())
+		return
+	}
+	if apiTeam, ok := apiResponse.Node.(*getNodeNodeTeam); ok {
+		teamOwnerUuid = apiTeam.Uuid
+	}
+
+	payload["name"] = plan.Name.ValueString()
+	payload["default_branch"] = plan.DefaultBranch.ValueString()
+	payload["show_api_token"] = true
+	payload["team_ids"] = []string{
+		teamOwnerUuid,
+	}
+
+	url := fmt.Sprintf("/v2/analytics/organizations/%s/suites", ts.client.organization)
+	err = ts.client.makeRequest("POST", url, payload, &response)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to create test suite", err.Error())
+		return
+	}
+
+	// TODO: dont rely on this, add the ID to the REST response
+	response.GraphqlID = base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("Suite---%s", response.UUID)))
+
+	state.ApiToken = types.StringValue(response.ApiToken)
+	state.DefaultBranch = types.StringValue(response.DefaultBranch)
+	state.ID = types.StringValue(response.GraphqlID)
+	state.UUID = types.StringValue(response.UUID)
+	state.Name = types.StringValue(response.Name)
+	state.Slug = types.StringValue(response.Slug)
+	state.TeamOwnerId = plan.TeamOwnerId
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
```

Comment:
Any reason this isn't `&state`, which seems to be the standard way we do this? This applies to all CRUD methods as far as I can tell.

Reply:
Hmm that is probably a copy-paste from the `State.Get()` but not having to copy a struct around is good so I'll change it